workitem/workitem_repository.go

```diff
@@ -99,6 +119,120 @@ func (r *GormWorkItemRepository) Delete(ctx context.Context, ID string) error {
 	return nil
 }
 
+// Reorder places the to-be-reordered workitem(s) above the input workitem.
+// The order of workitems are spaced by a factor of 1000.
+// The new order of workitem := (order of previousitem + order of nextitem)/2
+// Version must be the same as the one int the stored version
+func (r *GormWorkItemRepository) Reorder(ctx context.Context, beforeID string, wi app.WorkItem) (*app.WorkItem, error) {
+	var order float64
+	res := WorkItem{}
+	beforeItem := WorkItem{}
+	afterItem := WorkItem{}
+	id, err := strconv.ParseUint(wi.ID, 10, 64)
+	if err != nil || id == 0 {
+		return nil, errors.NewNotFoundError("work item", wi.ID)
+	}
+
+	tx := r.db.First(&res, id)
+	if tx.RecordNotFound() {
+		return nil, errors.NewNotFoundError("work item", wi.ID)
+	}
+	if tx.Error != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	if res.Version != wi.Version {
+		return nil, errors.NewVersionConflictError("version conflict")
+	}
+
+	wiType, err := r.wir.LoadTypeFromDB(ctx, wi.Type)
+	if err != nil {
+		return nil, errors.NewBadParameterError("Type", wi.Type)
+	}
+	if beforeID != "" {
```

Comment:
@DhritiShikhar Can you maybe create small functions and use them in these huge `if` blocks? The complexity of this function would dramatically decrease and maintainability increases. What do you think?